### PR TITLE
feat: allow non-uuid data types for vectorstore primary key

### PR DESCRIPTION
**Pull request by dishaprakash**
feat: allow non-uuid data types for vectorstore primary key

### CODE REVIEW

File: src/langchain_google_cloud_sql_pg/engine.py [hunk 1]
```diff
@@ -381,9 +381,11 @@ async def ainit_vectorstore_table(
         id_column: str = "langchain_id",
         overwrite_existing: bool = False,
         store_metadata: bool = True,
+        id_data_type: Column = Column("langchain_id", "UUID", nullable=False),
```
Comment (averikitsch) [inline src/langchain_google_cloud_sql_pg/engine.py:384]: We should make the id_column field `Union[str, Column]`. Where if it's just a string, then configure just the column name, and if it's a Column configure the name, type, and nullable.

Comment (dishaprakash) [reply src/langchain_google_cloud_sql_pg/engine.py]: The change has been made

**Review by vishwarajanand — APPROVED** [review]
Lgtm

File: src/langchain_google_cloud_sql_pg/engine.py [hunk 2]
```diff
@@ -524,12 +529,13 @@ def init_vectorstore_table(
...
         metadata_columns: List[Column] = [],
         metadata_json_column: str = "langchain_metadata",
-        id_column: str = "langchain_id",
+        id_column: Union[str, Column] = "langchain_id",
         overwrite_existing: bool = False,
         store_metadata: bool = True,
     ) -> None:
         """
         Create a table for saving of vectors to be used with PostgresVectorStore.
+        If id column data type is invalid, an UNDEFINED_OBJECT_ERROR is thrown.
```
Comment (jackwotherspoon) [inline src/langchain_google_cloud_sql_pg/engine.py:538]: nit: This does not follow [Google's style guide for docstrings](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings). You have the `Raises` block that should cover the error , it should not be in the initial description.

**Review by averikitsch — CHANGES_REQUESTED** [review]
Please see comments from AlloyDB

File: src/langchain_google_cloud_sql_pg/engine.py [hunk 3]
```diff
@@ -544,11 +550,13 @@ def init_vectorstore_table(
...
                 Default: "langchain_metadata". Optional.
-            id_column (str):  Name of the column to store ids.
-                Default: "langchain_id". Optional,
+            id_column (Union[str, Column]) : Column to store ids.
+                Default: "langchain_id" column name with data type UUID. Optional.
             overwrite_existing (bool): Whether to drop existing table. Default: False.
             store_metadata (bool): Whether to store metadata in the table.
                 Default: True.
+        Raises:
+            :class:`UndefinedObjectError <asyncpg.exceptions.UndefinedObjectError>`: if id column data type is invalid.
```
Comment (jackwotherspoon) [inline src/langchain_google_cloud_sql_pg/engine.py:559]: nit: "invalid" is vague and does not help user understand what this error suggest. What is deemed invalid?
```suggestion
           UndefinedObjectError: if id column data type is invalid.
```

File: src/langchain_google_cloud_sql_pg/vectorstore.py [hunk 4]
```diff
@@ -187,10 +187,10 @@ async def aadd_texts(
         self,
         texts: Iterable[str],
         metadatas: Optional[List[dict]] = None,
-        ids: Optional[List[str]] = None,
+        ids: Optional[List] = None,
         **kwargs: Any,
     ) -> List[str]:
-        """Embed texts and add to the table."""
+        """Embed texts and add to the table. Throws an error if Id column data type doesn't match with the ids."""
```
Comment (jackwotherspoon) [inline src/langchain_google_cloud_sql_pg/vectorstore.py:193]: nit: I wonder if we can make this a bit more clear?
```suggestion
        """Embed texts and add to the table. 
        
        Throws an error if the `ids` data type does not match that of the `id_column`.
        """
```

Comment (jackwotherspoon) [reply src/langchain_google_cloud_sql_pg/vectorstore.py:193]: ditto to spots below

File: src/langchain_google_cloud_sql_pg/engine.py [hunk 5]
```diff
@@ -474,12 +477,13 @@ async def ainit_vectorstore_table(
...
         metadata_columns: List[Column] = [],
         metadata_json_column: str = "langchain_metadata",
-        id_column: str = "langchain_id",
+        id_column: Union[str, Column] = "langchain_id",
         overwrite_existing: bool = False,
         store_metadata: bool = True,
     ) -> None:
         """
         Create a table for saving of vectors to be used with PostgresVectorStore.
+        If id column data type is invalid, an UNDEFINED_OBJECT_ERROR is thrown.
```
Comment (averikitsch) [inline src/langchain_google_cloud_sql_pg/engine.py:487]:
```suggestion
        Create a table for saving of vectors to be used with PostgresVectorStore.

```

File: src/langchain_google_cloud_sql_pg/engine.py [hunk 6]
```diff
@@ -524,12 +528,13 @@ def init_vectorstore_table(
...
```
Comment (averikitsch) [inline src/langchain_google_cloud_sql_pg/engine.py:537]:
```suggestion
        Create a table for saving of vectors to be used with PostgresVectorStore.
```

File: src/langchain_google_cloud_sql_pg/engine.py [hunk 7]
```diff
@@ -544,11 +547,13 @@ def init_vectorstore_table(
...
                 Default: "langchain_metadata". Optional.
-            id_column (str):  Name of the column to store ids.
-                Default: "langchain_id". Optional,
+            id_column (Union[str, Column]) : Column to store ids.
+                Default: "langchain_id" column name with data type UUID. Optional.
             overwrite_existing (bool): Whether to drop existing table. Default: False.
             store_metadata (bool): Whether to store metadata in the table.
                 Default: True.
+        Raises:
+            :class:`UndefinedObjectError <asyncpg.exceptions.UndefinedObjectError>`: if id column data type is invalid.
```
Comment (averikitsch) [inline src/langchain_google_cloud_sql_pg/engine.py:556]:
```suggestion
            :class:`UndefinedObjectError <asyncpg.exceptions.UndefinedObjectError>`: if the `ids` data type does not match that of the `id_column`.
```

**Review by averikitsch — APPROVED** [review]

**Pull request merged**